packages/uniforms/src/JSONSchemaBridge.js

```diff
@@ -0,0 +1,229 @@
+import invariant from 'fbjs/lib/invariant';
+import joinName  from './joinName';
+
+import Bridge from './Bridge';
+
+let Ajv;
+try {
+    const r = require; // Silence Meteor missing module warning
+    Ajv = r('ajv');
+} catch (_) { /* Ignore it. */ }
+
+const resolveRef = (referance, schema) => {
+    invariant(
+        referance.startsWith('#'),
+        'Reference is not an internal reference, and only such are allowed: "%s"',
+        referance
+    );
+
+    const resolvedReference = referance
+        .split('/')
+        .filter(part => part && part !== '#')
+        .reduce((definition, next) => definition[next], schema);
+
+    invariant(resolvedReference, 'Reference not found in schema: "%s"', referance);
+
+    return resolvedReference;
+};
+
+export default class JSONSchemaBridge extends Bridge {
+    constructor (schema) {
+        super();
+
+        invariant(schema.properties, 'Schema does not contain properties field');
```

Comment:
How about moving it to `check`? Constructor can assume, that schema is valid (at least, other schemas do).